File: VibgyorPaints/src/app/core/services/feedback.service.ts
import { Injectable } from '@angular/core';
import { Feedback } from '../models/feedback';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { environment } from 'src/environments/environment.prod';

const httpOptions = {
  headers: new HttpHeaders({'Content-Type': 'application/json'})
};

@Injectable({
  providedIn: 'root'
})
export class FeedbackService {

  constructor(private http: HttpClient) {}

  getFeedbacks() {
    return this.http.get(environment.DATA_URL + 'feedbacks');
  }

  getFeedback(id: number) {
    return this.http.get(environment.DATA_URL + 'feedback/' + id);
  }

  create(feedback: Feedback) {
    const body = JSON.stringify(feedback);
    return this.http.post(environment.DATA_URL + 'feedback', body, httpOptions);
  }

  update(feedback: Feedback) {
    const body = JSON.stringify(feedback);
    return this.http.put(environment.DATA_URL + 'feedback/' + feedback.id, body, httpOptions);
  }
}

File: VibgorPaintsApi/src/main/java/com/vibgyorpaints/service/FeedbackService.java
package com.vibgyorpaints.service;

import java.util.List;

import com.vibgyorpaints.model.Feedback;

public interface FeedbackService {

	List<Feedback> getFeedbacks();

	Feedback createFeedback(Feedback Feedback);

	Feedback getFeedback(Long id);
	
	Feedback update(Long id, Feedback Feedback);
	
}